src/domain/entities/notePublic.ts

```diff
@@ -0,0 +1,42 @@
+import type User from '@domain/entities/user.js';
+import type NoteSettingsPublic from './noteSettingsPublic';
+
+/**
+ * Note entity
+ */
+export interface NotePublic {
```

Comment:
You don't need to redeclare fields that stays the same. Describe only changed properties. And its better to explicitly describe public properties.

Something like:


```ts
type notePublicProperties = "content" | "createdAt" | "updatedAt";

interface NotePublic extends Pick<Note, notePublicProperties>{
  /**
   * Expose public id as the "id" property
   */
  id: NotePublicId;
}
```

Reply:
still actual

Reply:
changed